### DISCUSSION TITLE
Generating Lables

### DISCUSSION THREAD

**Discussion by Shuario** [snipe/snipe-it#9999] [Q&A]
Hi,

We bought Zebra printer ZD421-300dpi which is recommended but we are having trouble setting up right size of the labels. So I wonder if  there is anyone using same or similiar printer and could possibly give us hepling hand with setting up lable properties.

M

**Comment by Shiboneko** [comment]
Hi,
We bought a Zebra GK420d extra for the purpose of printing labels for our assets. But during my tests I found out, that the printing result is not very good, when done from inside the browser. I think this is caused by the driver or something, because the driver tries to somehow transform the label shown into a printable document, even though the printer itself has a function to generate QR codes.

Additionally had problems setting up the labels in snipe and also on the local pc. Thinking about, how everyone of my colleagues should be able to print "pristine" labels for out assets and sharing those settings would be a small nightmare.

This is when I decided to write a powershell script, which sends a ZPLII document, designed in ZDesigner, directly to the printer.
Much better result and more flexibility for me.

Though, I would love if I could hook into snipe, to make the printing simpler. Right now, I have to open powershell, copy the asset id and then use that argument in my powershell function. This might be too much for my non-technical colleagues.
I'd love to trigger a webapi call when clicking on the "generate Label" button in snipe.

**Reply by Shuario** [reply]
Hi,

would you mind sharing  PS script you are using?

**Reply by Shiboneko** [reply]
Hi,
it took some days, as I was busy with some other things.
I've put my example in a repository

You can grab it here: https://github.com/Shiboneko/Snipeitps.label

Once v6 came out, then I might try to put that code into a php file, which the server would call, when clicking the *generate label* button
But for now, this is my solution, to print labels at my company.
